src/avatar.jsx

```diff
@@ -156,7 +154,7 @@ const Avatar = React.createClass({
           className={className}
         >
           {icon && React.cloneElement(icon, {
-            color: color,
+            color: color || avatar.color,
```

Comment:
Good discussion :+1: @newoga It might not be a bad idea to open an issue for that. I have also noticed this issue. and it's not pretty at all.